cpp/CMakeLists.txt

```diff
@@ -455,11 +455,14 @@ if (UNIX)
   message(STATUS "Found cpplint executable at ${CPPLINT_BIN}")
 
   # Full lint
-  add_custom_target(lint ${CPPLINT_BIN}
+  # Balancing act: cpplint.py takes a non-trivial time to launch,
+  # so process 12 files per invocation, while still ensuring parallelism
+  add_custom_target(lint echo ${FILTERED_LINT_FILES} | xargs -n12 -P8
```

Comment:
This probably won't work on Windows but I guess we don't execute this on windows? Otherwise we could turn this into several targets that all form a single lint target at the end and let make/ninja handle the parallelisation.

Reply:
The lint target is only being created on non-Windows at the moment. I opened https://issues.apache.org/jira/browse/ARROW-2112 to make that possible

Reply:
Indeed there's a `if (UNIX)` a few lines above.